src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -238,6 +238,42 @@ function ishermitian_matrix(B::MatElem{T}) where T <: FinFieldElem
    return true
 end
 
+# return (true, h) if y = hx, (false, nothing) otherwise
+# FIXME: at the moment, works only for fields
+function _is_scalar_multiple_mat(x::MatElem{T}, y::MatElem{T}) where T <: RingElem
+   F=base_ring(x)
+   F==base_ring(y) || return (false, nothing)
+   nrows(x)==nrows(y) || return (false, nothing)
+   ncols(x)==ncols(y) || return (false, nothing)
+
+   if x==0
+      if y==0 return (true, F(1))
```

Comment:
To check that those matrices are zero matrices, in general one has to scan every single entry; so now we are doing that twice.

With the modified loop below, this becomes unnecessary.